libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1582,7 +1568,8 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 		if (memberFunctionType->kind() == FunctionType::Kind::Internal)
 		{
 			define(IRVariable(_memberAccess).part("functionIdentifier")) << to_string(functionDefinition.id()) << "\n";
-			m_context.internalFunctionAccessed(_memberAccess, functionDefinition);
+			if (!_memberAccess.annotation().calledDirectly)
+				m_context.internalFunctionAccessed(_memberAccess, functionDefinition);
```

Comment:
I see that the number of cases where `internalFunctionAccessed()` is called in `endVisit(MemberAccess)` grew significantly from the last time I touched this code. I think that it might be a good idea to replace them all with a single `if` at the top of the function that just calls `internalFunctionAccessed()` if the member access resolves to a function. Otherwise it's easy to miss one we didn't think of.

I was recently bitten in a similar way by `(L.f)()`. It broke my assumption that library functions can never go through the internal dispatch because we can't assign them to function pointers.